File: src/main/java/org/example/chessplatformbe/persistence/ArticleRepository.java
package org.example.chessplatformbe.persistence;

import org.example.chessplatformbe.domain.Article;
import org.example.chessplatformbe.exceptions.InvalidArticleException;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Page;


import java.util.Optional;

public interface ArticleRepository {
    Optional<Article> findById(Integer id);

    Article save(Article article);

    void delete(Article article);

    Page<Article> getArticlePage(Pageable pageable);

    Article update(Article article, Integer updateId);

    Article findByTitle(String title) throws InvalidArticleException;

    Page<Article> getArticlesByTitlePage(String title, Pageable pageable);
}

File: src/persistence/article_repository.rs
use crate::domain::{article::Article, page::Page, pageable::Pageable};

pub trait ArticleReporitory {
    fn find_by_id(&self, id: u32) -> Option<Article>;
    fn save(&mut self, article: Article) -> Article;
    fn delete(&mut self, article: Article);
    fn get_article_page(&self, pageable: Pageable) -> Page<Article>;
    fn update(&mut self, article: Article, update_id: u32) -> Article;
    fn find_by_title(&self, title: &str) -> Article;
    fn get_articles_by_title_page(&self, pageable: Pageable) -> Page<Article>;
}
